Platform/Plugins/com.tle.platform.common/src/com/tle/common/util/ExecUtils.java

```
@@ -52,6 +54,104 @@
     "", ".exe", ".bat"
   }; //$NON-NLS-1$ //$NON-NLS-2$ //$NON-NLS-3$
 
+  /**
+   * For a given Process ID, kill any child processes and then kill the process. Works on Unix only,
+   * as it leverages pgrep and kill commands.
+   *
+   * @param pid The Process for which to terminate including it's direct children.
+   */
+  public static void killLinuxProcessTree(int pid) {
+    if (pid > 0) {
```

Rather than silently doing nothing, how about invert this logic:

```java
if(pid < 1) {
  throw new IllegalArgumentException("Process ID should be greater than 0");
}
```